src/test/java/org/example/bookstore/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,179 @@
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import javax.sql.DataSource;
+import org.example.bookstore.dto.book.BookDtoWithoutCategoryIds;
+import org.example.bookstore.dto.category.CategoryDto;
+import org.example.bookstore.dto.category.CreateCategoryRequestDto;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.core.io.ClassPathResource;
+import org.springframework.http.MediaType;
+import org.springframework.jdbc.datasource.init.ScriptUtils;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class CategoryControllerTest {
+    public static final String DELETE_DATA_FROM_DB = "sql/delete-data-from-db.sql";
+    public static final String INSERT_DATA_INTO_DB = "sql/insert-data-into-db.sql";
+
+    protected static MockMvc mockMvc;
+
+    private static final Long CATEGORY_ID = 1L;
+    private static final String CATEGORY_NAME = "Historical Fiction";
+    private static final String CATEGORY_DESCRIPTION =
+            "A genre of fiction that takes place in the past.";
+    private static final Long INVALID_CATEGORY_ID = 999L;
+    private static final Long EXPECTED_CATEGORY_ID = 6L;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(@Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @BeforeEach
+    void setUp(@Autowired DataSource dataSource) throws Exception {
+        try (var connection = dataSource.getConnection()) {
+            connection.setAutoCommit(true);
+            ScriptUtils.executeSqlScript(connection, new ClassPathResource(DELETE_DATA_FROM_DB));
+            ScriptUtils.executeSqlScript(connection, new ClassPathResource(INSERT_DATA_INTO_DB));
+        }
+    }
```

Comment:
Please be consistent and use SQL annotation to run scripts before and after each test